lib/vector-buffers/src/topology/channel/sender.rs

```diff
@@ -13,6 +13,39 @@ use crate::{
     Bufferable, WhenFull,
 };
 
+/// Error related to sending an item via `BufferSender`.
+///
+/// This error type is meant to opaquely wrap the various error types returned by different buffer
+/// implementations in order to pass along the underlying error message to higher-level callers.
+#[derive(Debug)]
+pub struct SendError {
+    inner: Box<dyn error::Error + Send + Sync>,
+}
+
+impl SendError {
+    fn new(inner: impl error::Error + Send + Sync + 'static) -> Self {
+        Self {
+            inner: Box::new(inner),
+        }
+    }
+}
```

Comment:
Should this maybe be an `impl From`?

Reply:
Trying to use `impl<E> From<Box<E>> for SendError where E: std::error::Error ...`, among other patterns, does not seem to jive with implicit conversions. That was actually my initial approach.

I might take another crack at it, given we do roughly the same thing in a lot of other places, but I remember the compiler being incredibly angry about this.

Reply:
Resolved by removing `SendError` entirely in ddd91a8d6.